EpikCord/__init__.py

```diff
@@ -1156,6 +1113,151 @@ async def command_error(
+    async def guild_update(self, data) -> tuple[Optional[Guild], Guild]:
+        after:Guild = Guild(self, data)
+        before:Optional[Guild] = self.guilds.fetch(after.id)
+        return before ,after
+    
+    async def guild_delete(self, data: dict) -> Optional[Guild]:
+        return self.guilds.get(data["id"])
+
+    async def guild_ban_add(self, data:dict) -> tuple[Optional[Guild], User]:
+        return self.guilds.fetch(data["guild_id"]), User(self, data["user"])
+
+    async def guild_ban_remove(self, data:dict)-> tuple[Optional[Guild], User]:
+        return self.guilds.fetch(data["guild_id"]), User(self, data["user"])
+    
+    async def guild_emojis_update(self,data:dict) -> tuple[Optional[Guild], List[Emoji]]:
+        emojis = [Emoji(self,emoji, data["guild_id"]) for emoji in data["emojis"]]
+        return self.guilds.fetch(data["guild_id"]), emojis
+
+    async def guild_stickers_update(self,data:dict) -> tuple[Optional[Guild], List[Sticker]]:
+        stickers = [Sticker(sticker) for sticker in data["stickers"]]
+        return self.guilds.fetch(data["guild_id"]), stickers 
+
+    async def guild_integrations_update(self,data:dict) -> Optional[Guild]:
+        return self.guilds.fetch(data["guild_id"])
+
+    async def guild_member_add(self,data:dict) -> tuple[Optional[Guild], GuildMember]:
+        return self.guilds.fetch(data["guild_id"]), GuildMember(self, data)    
+
+    async def guild_member_update(self, data) -> tuple[Optional[GuildMember],GuildMember]:
+        guild_member = GuildMember(self, data)
+        return self.members.fetch(data["id"]), guild_member
+
+    async def guild_member_remove(self, data:dict) -> tuple[Optional[Guild], User]:
+        return self.guilds.fetch(data["guild_id"]), User(self,data["user"])
+
+    async def guild_members_chunk(self, data: dict):
+        ...
+    
+    # Guild Role: TODO: Cache Roles per Server
+    async def guild_role_create(self,data:dict) -> Role:
+        role = Role(self,data["role"])
+        self.roles.add_to_cache(role.id, role)
+        return role
+
+    async def guild_role_update(self,data:dict) -> tuple[Optional[Role], Role]:
+        after = Role(self, data["role"])
+        before = self.roles.get(after.id)
+        return before, after 
+
+    async def guild_role_delete(self, data:dict) -> Optional[Role]:
+        deleted = self.roles.get(data["role_id"])
+        return deleted
+    
+    # Guild Scheduled Event Create: TODO: Caching
+    async def guild_scheduled_event_create(self, data:dict) -> GuildScheduledEvent:
+        event = GuildScheduledEvent(self, data)
+        self.guild_events.add_to_cache(event.id, event)
+        return event
+    
+    async def guild_scheduled_event_update(self,data:dict):
+        after = GuildScheduledEvent(self,data)
+        before = self.guild_events.get(after.id)
+        return before, after
+    async def guild_scheduled_event_delete(self,data:dict):
+        return GuildScheduledEvent(self, data)
```

Comment:
Are you sure they return the entire GuildScheduledEvent in the data?

Reply:
YES